CHANGELOG.md

```diff
@@ -53,6 +53,147 @@ This name should be decided amongst the team before the release.
 * [#535](https://github.com/tweag/topiary/pull/535) Improved error message when idempotency fails due to invalid output in the first pass.
 * [#533](https://github.com/tweag/topiary/pull/533) Update tree-sitter-ocaml to 0.20.3
 * [#576](https://github.com/tweag/topiary/pull/576) Allows prepending/appending `@begin_scope` and `@end_scope`
+* [#583](https://github.com/tweag/topiary/pull/583) Modernisation of the command line interface (see [below](#cli-migration-guide), for details)
+
+#### CLI Migration Guide
+
+Full documentation for the CLI can be found in the project's
+[`README`](/README.md). Herein we summarise how the v0.2.3 functionality
+maps to the new interface, to aid migration.
+
+##### Formatting
+
+###### From Files, In Place
+
+Before:
+```
+topiary [--skip-idempotence] \
+        [--tolerate-parsing-errors] \
+        --in-place \
+        --input-files INPUT_FILES...
+```
+
+After:
+```
+topiary fmt [--skip-idempotence] \
+            [--tolerate-parsing-errors] \
+            INPUT_FILES...
+```
+
+###### From File, To New File
+
+Before:
+```
+topiary [--skip-idempotence] \
+        [--tolerate-parsing-errors] \
+        (--langauge LANGUAGE | --query QUERY) \
+        --input-files INPUT_FILE \
+        --output-file OUTPUT_FILE
+```
+
+After (use IO redirection):
+```
+topiary [--skip-idempotence] \
+        [--tolerate-parsing-errors] \
+        (--langauge LANGUAGE [--query QUERY]) \
+        < INPUT_FILE \
+        > OUTPUT_FILE
+```
+
+###### Involving Standard Input and Output
```

Comment:
It's to be more explicit about the case when, before, `-` was interpreted as stdin/stdout, rather than a filename (should anyone want to use `-` as a filename).